ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -1082,6 +1171,55 @@ func validateTaskNetworkConfig(taskNetworkConfig *state.TaskNetworkConfig) error
 	return nil
 }
 
+// startNetworkLatencyFault invokes the linux TC utility tool to start the network-latency fault.
+func (h *FaultHandler) startNetworkLatencyFault(ctx context.Context, taskMetadata *state.TaskResponse, request types.NetworkLatencyRequest) error {
+	interfaceName := taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].NetworkInterfaces[0].DeviceName
+	networkMode := taskMetadata.TaskNetworkConfig.NetworkMode
+	// If task's network mode is awsvpc, we need to run nsenter to access the task's network namespace.
+	nsenterPrefix := ""
+	if networkMode == ecs.NetworkModeAwsvpc {
+		nsenterPrefix = fmt.Sprintf(nsenterCommandString, taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].Path)
+	}
+	delayInMs := aws.Uint64Value(request.DelayMilliseconds)
+	jitterInMs := aws.Uint64Value(request.JitterMilliseconds)
+
+	// Command to be executed:
+	// <nsenterPrefix> tc qdisc add dev <interfaceName> root handle 1: prio priomap 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
+	// <nsenterPrefix> "tc qdisc add dev <interfaceName> parent 1:1 handle 10: netem delay <latency>ms <jitter>ms
+	tcAddQdiscRootCommandComposed := nsenterPrefix + fmt.Sprintf(tcAddQdiscRootCommandString, interfaceName)
+	cmdList := strings.Split(tcAddQdiscRootCommandComposed, " ")
+	cmdOutput, err := h.runExecCommand(ctx, cmdList)
+	if err != nil {
+		logger.Error(fmt.Sprintf("'%s' command failed with the following error: '%s'. std output: '%s'. TaskArn: %s",
```

Comment:
nit(non-blocking): might make it a bit easier to read if we use `logger.Field`

Reply:
ditto for the rest of the logger statements (non-blocking/up to you)

Reply:
SGTM. Will refactor the logs in the helper methods